nextmv/nextmv/cloud/application.py

```diff
@@ -1048,6 +1297,124 @@ def new_run_with_result(
+        id: str,
+        name: str,
+        scenarios: list[Scenario],
+        description: Optional[str] = None,
+        repetitions: Optional[int] = 0,
+    ) -> str:
+        """
+        Create a new scenario test. The test is based on `scenarios` and you
+        may specify `repetitions` to run the test multiple times. 0 repetitions
+        means that the tests will be executed once. 1 repetition means that the
+        test will be repeated once, i.e.: it will be executed twice. 2
+        repetitions equals 3 executions, so on, and so forth.
+
+        For each scenario, consider the `scenario_input` and `configuration`.
+        The `scenario_input.scenario_input_type` allows you to specify the data
+        that will be used for that scenario.
+
+        - `ScenarioInputType.INPUT_SET`: the data should be taken from an
+          existing input set.
+        - `ScenarioInputType.INPUT`: the data should be taken from a list of
+          existing inputs. When using this type, an input set will be created
+          from this set of managed inputs.
+        - `ScenarioInputType.New`: a new set of data will be uploaded as a set
+          of managed inputs. A new input set will be created from this set of
+          managed inputs.
+
+        On the other hand, the `configuration` allows you to specify multiple
+        option variations for the scenario. Please see the
+        `ScenarioConfiguration` class for more information.
+
+        The scenario tests uses the batch experiments API under the hood.
+
+        Parameters
+        ----------
+        id: str
+            ID of the scenario test.
+        name: str
+            Name of the scenario test.
+        scenarios: list[Scenario]
+            List of scenarios to use for the scenario test. At least one
+            scenario should be provided.
+        description: Optional[str]
+            Optional description of the scenario test.
+        repetitions: Optional[int]
+            Number of repetitions to use for the scenario test. 0
+            repetitions means that the tests will be executed once. 1
+            repetition means that the test will be repeated once, i.e.: it
+            will be executed twice. 2 repetitions equals 3 executions, so on,
+            and so forth.
+
+        Returns
+        -------
+        str
+            ID of the scenario test.
+
+        Raises
+        ------
+        requests.HTTPError
+            If the response status code is not 2xx.
+        ValueError
+            If no scenarios are provided.
+        """
+
+        if len(scenarios) < 1:
+            raise ValueError("At least one scenario must be provided")
+
+        # Save all the information needed by scenario.
+        input_sets = {}
+        scenarios_by_id = {}
+        instances = {}
+        for scenario_ix, scenario in enumerate(scenarios):
+            scenario_id = f"scenario-{scenario_ix}" if scenario.scenario_id is None else scenario.scenario_id
```

Comment:
Do we need to verify that we are generating a unique ID here? I.e., `scenario.scenario_id` of one scenario could technically already be defined as "scenario-0" which we would generate for the first if it wouldn't have an ID. :thinking: 

Probably a bit overcautious for now. I only use scenario testing occasionally and don't have a good sense for the use case of this `nextmv-py` functionality yet. :blush:

Reply:
Good point. Better be safe than sorry. I will create a function for handling scenario IDs creation and checking that they result in unique keys.

Reply:
I created the `_scenarios_by_id` that creates the IDs of the scenarios and makes sure that they are unique.